### DISCUSSION TITLE
Banning based on forwarded IP

### DISCUSSION THREAD

**Discussion by stashymane** [tomMoulard/fail2ban#83] [Q&A] [answered]
If I have my site behind a Cloudflare proxy and set up a [Cloudflare IP middleware](https://github.com/BetterCorp/cloudflarewarp) to output the actual visitor's IP to the X-Real-IP header, does fail2ban use the forwarded IP or the one from Cloudflare? I couldn't find any documentation regarding this yet, so help is appreciated, thanks :)

**Comment by tomMoulard** [comment, marked as answer]
Hello @stashymane,

Thanks for your interest in this Traefik plugin !

For now, this plugin only looks for the request IP, and it does not use the `X-Forwarded-For` header (yet).

**Reply by tomMoulard** [reply]
This is linked to this behavior:

https://github.com/tomMoulard/fail2ban/blob/2790061e862b27fb62a992822cd6d9bd2beed887/fail2ban.go#L220

**Comment by tajchert** [comment]
This would be great addition. As in Go RemoteAddr is (correctly) a last IP address - so in most cases it might be some router/loadbalance (if there any), or... as in my and @stashymane  case a Cloudflare tunnel.

I tried to create a chain of 2 middlewares but failed, this might require change in the code (maybe being able to customize on which header would be the best and most versatile solution?).
